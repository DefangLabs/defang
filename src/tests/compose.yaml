name: tests
services:
  dfnx:
    build:
      context: .
      dockerfile: Dockerfile
      args:
        DNS: "${COMPOSE_PROJECT_NAME}-dfnx.prod1.defang.dev"
    env_file:
      - fileName.env
    deploy:
      resources:
        limits:
          cpus: '0.50'
          memory: 512M
        reservations:
          cpus: '0.25'
          memory: 256M
    ports:
      - target: 80
        mode: ingress
      - target: 1234
        # mode: host
    secrets:
      - dummy
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
      # disable: true

  # dfnx:
  #   build:
  #     context: .
      # dockerfile: Dockerfile.dfn
    # ports:
      # - 80

  echo:
    image: ealen/echo-server
    ports:
      - target: 80
        mode: ingress
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost/"]
    # domainname: echotest.gnafed.click
    profiles:
      - donotstart

secrets:
  dummy:
    external: true
    name: dummy
