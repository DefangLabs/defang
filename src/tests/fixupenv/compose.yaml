services:
  Mistral:
    image: "mistral:latest"
    ports:
      - mode: host
        target: 8000
  ui:
    image: "ui:latest"
    environment:
      - "API_URL=http://Mistral:8000"
      - "SENSITIVE_DATA"
  bad-service:
    image: "somedb:latest"
    ports:
      - mode: ingress
        target: 5432
  use-bad-service:
    image: "service:latest"
    environment:
      - "DB_URL=bad-service:5432"
  env-in-config:
    image: "service:latest"
    environment:
      - "CONFIG1=http://Mistral:8000"
