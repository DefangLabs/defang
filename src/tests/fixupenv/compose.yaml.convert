[
  {
    "name": "bad-service",
    "image": "somedb:latest",
    "platform": 2,
    "internal": true,
    "ports": [
      {
        "target": 5432,
        "mode": 1
      }
    ],
    "networks": 1
  },
  {
    "name": "env-in-config",
    "image": "service:latest",
    "platform": 2,
    "internal": true,
    "environment": {
      "CONFIG1": "http://mistral:8000"
    },
    "networks": 1
  },
  {
    "name": "mistral",
    "image": "mistral:latest",
    "platform": 2,
    "internal": true,
    "ports": [
      {
        "target": 8000
      }
    ],
    "networks": 1
  },
  {
    "name": "ui",
    "image": "ui:latest",
    "platform": 2,
    "internal": true,
    "environment": {
      "API_URL": "http://mistral:8000"
    },
    "secrets": [
      {
        "source": "SENSITIVE_DATA"
      }
    ],
    "networks": 1
  },
  {
    "name": "use-bad-service",
    "image": "service:latest",
    "platform": 2,
    "internal": true,
    "environment": {
      "DB_URL": "bad-service:5432"
    },
    "networks": 1
  }
]