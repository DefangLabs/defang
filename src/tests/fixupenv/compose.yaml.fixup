[
  {
    "command": null,
    "entrypoint": null,
    "image": "somedb:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "ingress",
        "target": 5432,
        "protocol": "tcp",
        "app_protocol": "http"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "environment": {
      "CONFIG1": "http://mistral:8000"
    },
    "image": "service:latest",
    "networks": {
      "default": null
    }
  },
  {
    "build": {
      "context": ".",
      "dockerfile": "Dockerfile",
      "args": {
        "API_URL": "http://mistral:8000"
      }
    },
    "command": null,
    "entrypoint": null,
    "networks": {
      "default": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "image": "mistral:latest",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 8000,
        "protocol": "tcp"
      }
    ]
  },
  {
    "command": null,
    "entrypoint": null,
    "environment": {
      "API_URL": "http://mistral:8000",
      "SENSITIVE_DATA": null
    },
    "image": "ui:latest",
    "networks": {
      "default": null
    }
  },
  {
    "command": null,
    "entrypoint": null,
    "environment": {
      "DB_URL": "bad-service:5432"
    },
    "image": "service:latest",
    "networks": {
      "default": null
    }
  }
]