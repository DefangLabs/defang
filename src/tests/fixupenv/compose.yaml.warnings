 ! service "Mistral": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "bad-service": ingress port without healthcheck defaults to GET / HTTP/1.1
 ! service "bad-service": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "env-in-config": environment variable "CONFIG1" will use the 'defang config' value instead of adjusted service name
 ! service "env-in-config": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "fixup-args": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "fixup-args": service name was adjusted: build argument "API_URL" assigned value "http://mock-mistral:8000"
 ! service "ui": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "ui": service name was adjusted: environment variable "API_URL" assigned value "http://mock-mistral:8000"
 ! service "use-bad-service": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "use-bad-service": service name in the environment variable "DB_URL" was not adjusted, only references to other services with port mode set to 'host' will be fixed-up
 ! service name "Mistral" was normalized to "mistral"
 * Packaging the project files for fixup-args at .