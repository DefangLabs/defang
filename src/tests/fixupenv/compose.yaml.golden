name: fixupenv
services:
  Mistral:
    image: mistral:latest
    networks:
      default: null
    ports:
      - mode: host
        target: 8000
  bad-service:
    image: somedb:latest
    networks:
      default: null
    ports:
      - mode: ingress
        target: 5432
  env-in-config:
    environment:
      CONFIG1: http://Mistral:8000
    image: service:latest
    networks:
      default: null
  ui:
    environment:
      API_URL: http://Mistral:8000
      SENSITIVE_DATA: null
    image: ui:latest
    networks:
      default: null
  use-bad-service:
    environment:
      DB_URL: bad-service:5432
    image: service:latest
    networks:
      default: null
networks:
  default:
    name: fixupenv_default
